lib/Transport.js

```diff
@@ -84,10 +84,15 @@ class Transport extends Events {
     }
 
     ping() {
-        this.ws.ping();
-        this.timoutId = setTimeout(() => {
-            this.ping();
-        }, this.pingInterval);
+        try {
+            this.ws.ping();
+            this.timoutId = setTimeout(() => {
+                this.ping();
```

Comment:
do we catch the exception coming from thing async part also?

Reply:
No, exceptions can't travel without a promise. The point in having those lines inside the try to to stop them from executing if this.ws.ping(); throws an exception. That way it doesn't continue to try the ping. The ping will restart if reconnect() gets called.